### PR TITLE
Update pd.rules.yml

### PR DESCRIPTION
**Pull request by miaoqingli**
Modify 'pending_peer_region_count' to 'pending-peer-region-count'.Modify 'incorrect_namespace_region_count' to 'incorrect-namespace-region-count'. The metrics stored in prometheus were named like 'xx-xx-xx' instead of 'xx_xx_xx'.

### Check List

Tests

- Unit test
- Integration test
- Manual test (add detailed scripts or steps below)
- No code

Code changes

- Has configuration change
- Has HTTP API interfaces change (Don't forget to [add the declarative for API](https://github.com/tikv/pd/blob/master/docs/development.md#updating-api-documentation))
- Has persistent data change

Side effects

- Possible performance regression
- Increased code complexity
- Breaking backward compatibility

Related changes

- PR to update [`pingcap/docs`](https://github.com/pingcap/docs)/[`pingcap/docs-cn`](https://github.com/pingcap/docs-cn):
- PR to update [`pingcap/tidb-ansible`](https://github.com/pingcap/tidb-ansible):
- Need to cherry-pick to the release branch

### Release note

```release-note
Please add a release note.

If you don't think this PR needs a release note then fill it with None.
```

### CODE REVIEW

**Comment by miaoqingli** [comment]
Modify 'pending_peer_region_count' to 'pending-peer-region-count'.Modify 'incorrect_namespace_region_count' to 'incorrect-namespace-region-count'. The metrics stored in prometheus were named like 'xx-xx-xx' instead of 'xx_xx_xx'.

File: metrics/alertmanager/pd.rules.yml [hunk 1]
```diff
@@ -98,24 +98,24 @@ groups:
       summary: PD_down_peer_region_nums
 
   - alert: PD_incorrect_namespace_region_count
-    expr: (sum(pd_regions_status{type="incorrect_namespace_region_count"}) by (instance)  > 100) and (sum(etcd_server_is_leader) by (instance) > 0)
+    expr: (sum(pd_regions_status{type="incorrect-namespace-region-count"}) by (instance)  > 100) and (sum(etcd_server_is_leader) by (instance) > 0)
```
Comment (rleungx) [inline metrics/alertmanager/pd.rules.yml:101]: We have already deprecated the namespace.

**Comment by rleungx** [comment]
BTW, please sign off DCO.

File: metrics/alertmanager/pd.rules.yml (hunk 1 above)
Comment (nolouch) [inline metrics/alertmanager/pd.rules.yml:101]: I think this can be remove.

**Review by nolouch — APPROVED** [review]

**Pull request merged**